test/compilable/traits_hasAliasing.d

```diff
@@ -0,0 +1,108 @@
+module traits_hasAliasing;
+
+@safe pure nothrow @nogc:
+
+/** Verify behaviour of `__traits(hasAliasing, void*)` being same as `std.traits.hasAliasing`.
+ *
+ * Copied from Phobos `std.traits`.
+ */
+void test_hasAliasing()
+{
+    static assert(__traits(hasAliasing, void*));
+    static assert(!__traits(hasAliasing, void function()));
+
```

Comment:
Please add tests with templates and when a type is referencing, not only classes but also interfaces.

Reply:
Also add tests for associative arrays

Reply:
> Please add tests with templates and when a type is referencing, not only classes, but also interfaces.

Can you elaborate with sample code what you mean?

Reply:
```d
Interface I;
struct S1 { int a; I b; }
struct S2 { int a; immutable I b; }
```

Reply:
Tests added.

Reply:
Add template tests at https://github.com/dlang/dmd/pull/13040/commits/107eff1cc0d4cbac4deacb0745c8185224175c71.

Reply:
Are you satisfied now, @ljmf00?